Userland/Utilities/cat.cpp

```diff
@@ -37,13 +41,43 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
     TRY(Core::System::pledge("stdio"));
 
+    size_t line_count = 1;
     Array<u8, 32768> buffer;
     for (auto const& file : files) {
         while (!file->is_eof()) {
             auto const buffer_span = TRY(file->read_some(buffer));
-            out("{:s}", buffer_span);
+            if (show_lines) {
+                line_count = output_buffer_with_line_numbers(line_count, buffer_span);
+            } else {
+                out("{:s}", buffer_span);
+            }
         }
     }
 
     return files.size() != paths.size();
 }
+
+size_t output_buffer_with_line_numbers(size_t line_count, ReadonlyBytes buffer_span)
+{
+    auto buffer_itter = buffer_span.begin();
+    while (!buffer_itter.is_end()) {
+        // to know if the next itter is the end
+        auto curr_value = *buffer_itter;
+        buffer_itter++;
+
+        if (line_count == 1 && (buffer_itter == buffer_span.begin())) {
```

Comment:
`buffer_itter == buffer_span.begin()`
This condition is never true as you always increment `buffer_itter` on the previous line.

Reply:
This makes me have this _wrong_ result:
```
$cat -n README.md 
# SerenityOS and Ladybird
        1	
        2	SerenityOS is a graphical Unix-like operating system for x86-64 computers.
...
```